CHANGELOG.md

```diff
@@ -20,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Fixed issue where Executor `parallelism` option in a Tango settings file would be ignored.
 - Fixed a bug where the unique ID of a step that depends on a key-value of the result of another step could change if the name of the other step changes.
+- Fixed a bug where importing certain libraries (like torchmetrics) would mess with our exception handling because they set `sys.excepthook` for some reason. Now we always reset `sys.excepthook` after importing.
```

Comment:
@ananyahjha93 Why does torchmetrics do this?

Reply:
Digging into this a little more:

```python
import sys

print(sys.excepthook)  # <built-in function excepthook>

import torchmetrics  # 0.10.1

print(sys.excepthook)  # <function install.<locals>.excepthook at 0x12af553a0>
print(sys.excepthook.__module__)  # rich.traceback
```

Reply:
But `rich` isn't a direct dependency of torchmetrics... And it's not a direct dependency of any of its direct dependencies either.. (numpy, torch, packing, typing_extensions)..

Reply:
But torchmetrics tries to import a bunch of dependencies that are not listed as dependencies, so it could be any of those.

Reply:
I don't know either. But maybe they overwrite the except hook for a reason, and we're inviting trouble if we undo it?

Reply:
I doubt that. The hook that's being set is this: https://rich.readthedocs.io/en/stable/traceback.html#traceback-handler

All it does is print a very verbose but pretty looking traceback. And in general I don't think we need to worry about not letting other libraries set the excepthook. I mean, what's the worst that could happen? It's only called when the program is crashing anyway...

Reply:
Other libraries could be cleaning up resources, or do something else important.

I'm uneasy about this. Can we detect that it's `rich`, and only reset when that's the case?

Reply:
Yeup: 9a6cd92